servers/dsub/jobs/controllers/jobs_controller.py

```diff
@@ -129,14 +129,21 @@ def query_jobs(body):
             microsecond=0)
         if create_time_max and query.start > create_time_max:
             raise BadRequest(
-                "Invalid pagination token with query start parameter.")
+                "Invalid query: start date is invalid with pagination token.")
+    if query.end:
+        query.end = query.end.replace(tzinfo=tzlocal()).replace(microsecond=0)
+        if query.start and query.start >= query.end:
+            raise BadRequest(
+                "Invalid query: start date must precede end date.")
 
     job_generator = _generate_dstat_jobs(provider, query, create_time_max,
                                          offset_id)
     jobs = []
     try:
         for job in job_generator:
-            jobs.append(job)
+            if not query.end or job.get(
```

Comment:
nit: I would probably invert and `continue` instead, which should set up better for additional filter criteria.

Reply:
Done (in generator function)